cobalt/base/process/process_metrics_helper_test.cc

```diff
@@ -0,0 +1,199 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/base/process/process_metrics_helper.h"
+
+#include <atomic>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/bind.h"
+#include "base/containers/contains.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+#include "base/threading/thread.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace base {
+
+namespace {
+
+static std::atomic<bool> keep_busy{true};
+
+void BusyWork(std::vector<std::string>* vec) {
+  int64_t test_value = 0;
+  while (keep_busy.load()) {
+    ++test_value;
+    vec->push_back(NumberToString(test_value));
+    if (vec->size() > 100000) vec->clear();
+  }
+}
+
+}  // namespace
+
+class ProcessMetricsHelperTest : public testing::Test {
+ public:
+  static void SetUpTestSuite() {
+    ProcessMetricsHelper::PopulateClockTicksPerS();
+  }
+
+  ProcessMetricsHelper::Fields GetProcStatFields(
+      ProcessMetricsHelper::ReadCallback cb,
+      std::initializer_list<int> indices) {
+    return ProcessMetricsHelper::GetProcStatFields(std::move(cb), indices);
+  }
+
+  int GetClockTicksPerS(ProcessMetricsHelper::ReadCallback uptime_callback,
+                        ProcessMetricsHelper::ReadCallback stat_callback) {
+    return ProcessMetricsHelper::GetClockTicksPerS(std::move(uptime_callback),
+                                                   std::move(stat_callback));
+  }
+
+  TimeDelta GetCPUUsage(ProcessMetricsHelper::ReadCallback stat_callback,
+                        int ticks_per_s) {
+    return ProcessMetricsHelper::GetCPUUsage(std::move(stat_callback),
+                                             ticks_per_s);
+  }
+};
+
+ProcessMetricsHelper::ReadCallback GetNulloptCallback() {
+  return BindOnce(
+      []() -> absl::optional<std::string> { return absl::nullopt; });
+}
+
+ProcessMetricsHelper::ReadCallback GetUptimeCallback() {
+  return BindOnce([]() -> absl::optional<std::string> {
+    return "1635667.97 155819443.97";
+  });
+}
+
+ProcessMetricsHelper::ReadCallback GetStatCallback() {
+  return BindOnce([]() -> absl::optional<std::string> {
+    return "3669677 (name with )( ) R 477564 3669677 477564 34817 3669677 "
```

Comment:
Can you add a comment here what this string means, and maybe a link to where there is some documentation about it?